src/stores/Inventory.js

```diff
@@ -0,0 +1,47 @@
+import { observable, computed, action, makeObservable } from 'mobx'
+import { Item } from './Item'
+
+
+export class Inventory {
+    constructor() {
+        this.list = [{ name: 'phone', price: 100, quantity: 7 }]
+
+        makeObservable(this, {
+            list: observable,
+            addItem: action,
+            changePrice: action,
+            buyItem: action,
+            numItems: computed
+        })
+    }
+
+    addItem = (name, price = 0, quantity = 1) => {
+        if (this.list.some(obj => obj.name === name)) {
+            let itemToChange = this.list.find(i => i.name === name)
+            itemToChange.price = price
+            itemToChange.quantity++
+        } else {
+            let newI = new Item(name, price, quantity)
+            this.list.push(newI)
+        }
+    }
+    changePrice = (name, newPrice) => {
+        let item = this.list.find(i => i.name === name)
+        item.price = newPrice
+    }
+    buyItem = (name) => {
+        let relevantItem = this.list.find(i => i.name === name)
```

Comment:
You should use `findIndex` here, that way you don't need to use `indexOf` (which shouldn't be used with objects in any case).